### ISSUE TITLE
Compute approximation custom feedback

### ISSUE THREAD

**Issue by TomasBlazek1** [GeoMop/bgem#15]
`compute_adaptive_approximation` is an iterative process which means it would be useful to define a callback function for showing progress of computation.

**Comment by jbrezmorf** [comment]
Good point. However, it would be better to run the computation in a separate thread and process its logging output.

**Comment by jbrezmorf** [comment]
Resolved by #20.

**Comment by jbrezmorf** [comment]
#20 should not be implemented and does not in fact solve the problem. We have to define a thread-safe mean that provides information about running task.
